### ISSUE TITLE
macOS aarch64-darwin (M1/M2): document since when Nixpkgs R versions are supported (?)

### ISSUE THREAD

**Issue by philipp-baumann** [b-rodrigues/rix#117]
<img width="962" alt="Screenshot 2024-01-29 at 20 44 31" src="https://github.com/b-rodrigues/rix/assets/21625034/aa76a333-0cf1-4f60-a9a5-eddff2b05984">

**Comment by philipp-baumann** [comment]
This fails for:

```{r}
path_subshell <- tempdir()

  rix_init(
    project_path = path_subshell,
    rprofile_action = "overwrite",
    message_type = "simple"
  )

 rix(
    r_ver = "3.5.3",
    overwrite = TRUE,
    project_path = path_subshell
  )

nix_build(project_path = path_subshell)
```

**Comment by b-rodrigues** [comment]
Could it be because that old release of R was not available for apple silicon architectures?

**Comment by philipp-baumann** [comment]
> Could it be because that old release of R was not available for apple silicon architectures?

maybe something as simple as that. i think it sounds like this is the problem https://github.com/NixOS/nixpkgs/pull/216992/files#diff-ad7b67d4a92a2db43281c9c45aa74c32341a554fbca0b6b25be593f4648abb5e

**Comment by philipp-baumann** [comment]
i had to add `bintools` and now it works on macOS :-) Let's tweak that in rix.

**Comment by philipp-baumann** [comment]
@b-rodrigues there is a bunch of different issues R versions before 2021 when M1/M2 (aarch64-darwin) wasn't yet supported in Nix, which makes sense. I tested, it works for R version 4.1.2 and above.

```sh
# default.nix
# This file was generated by the {rix} R package v0.5.1.9000 on 2024-01-29
# with following call:
# >rix(r_ver = "7592790b9e02f7f99ddcb1bd33fd44ff8df6a9a7",
#  > project_path = path_subshell,
#  > overwrite = TRUE)
# It uses nixpkgs' revision 7592790b9e02f7f99ddcb1bd33fd44ff8df6a9a7 for reproducibility purposes
# which will install R version 4.1.2
# Report any issues to https://github.com/b-rodrigues/rix
let
 pkgs = import (fetchTarball "https://github.com/NixOS/nixpkgs/archive/7592790b9e02f7f99ddcb1bd33fd44ff8df6a9a7.tar.gz") {};
    system_packages = builtins.attrValues {
  inherit (pkgs) R glibcLocales nix ;
};
  in
  pkgs.mkShell {
    LOCALE_ARCHIVE = if pkgs.system == "x86_64-linux" then  "${pkgs.glibcLocales}/lib/locale/locale-archive" else "";
    LANG = "en_US.UTF-8";
    LC_ALL = "en_US.UTF-8";
    LC_TIME = "en_US.UTF-8";
    LC_MONETARY = "en_US.UTF-8";
    LC_PAPER = "en_US.UTF-8";
    LC_MEASUREMENT = "en_US.UTF-8";

    buildInputs = [    system_packages  ];
      shellHook = "R --vanilla";
  }
```

**Comment by philipp-baumann** [comment]
![Uploading Screenshot 2024-01-29 at 23.09.08.png…]()

**Comment by philipp-baumann** [comment]
<img width="670" alt="Screenshot 2024-01-29 at 23 09 08" src="https://github.com/b-rodrigues/rix/assets/21625034/f8fa7016-c3d3-4d9c-bc1a-b8dcd60c2792">

**Comment by b-rodrigues** [comment]
yes, we should definitely add that to the macos vignette

**Comment by philipp-baumann** [comment]
solved with #149
